compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2349,6 +2353,43 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
             }
         }
     }
+
+    fn check_instruction_set(&self, attr: &Attribute, _item: Option<ItemLike<'_>>) {
```

Comment:
I think we also want to valid that `item` is a function or closure and not anything else like a module or type (and use on any of those should give a specific error that the attribute can only be used on functions and such).

Reply:
@wesleywiser 
Thank you for your review!! Ok, i'll fix that.